ipaserver/plugins/baseldap.py

```diff
@@ -1937,7 +1937,23 @@ def get_attr_filter(self, ldap, **options):
         """
         search_kw = self.args_options_2_entry(**options)
         search_kw['objectclass'] = self.obj.object_class
-        return ldap.make_filter(search_kw, rules=ldap.MATCH_ALL)
+        default_kw = self.get_default(**options)
+
+        filters = []
+        for name, value in search_kw.items():
+            if name == 'nsaccountlock' and value == False:
+                # If nsaccountlock is False, set the value to True and
+                # search for nsaccountlock != True. This way we can search
+                # for any users where nsaccountlock=False or nsaccountlock=None.
```

Comment:
OK, figured it out:
* The `nsaccountlock` param in `user` plugin does not have a default value specified.
* The check for default value below should actually look like this:
  ```python
              default = self.get_default_of(name, **options)
              if default is not None and value == default:
                  fltr = ldap.combine_filters([fltr, '(!({}=*))'.format(name)])
  ```